### DISCUSSION TITLE
Shading on a PV module

### DISCUSSION THREAD

**Discussion by Naveen-learner** [pvlib/pvlib-python#1321] [General]
How can monitoring data be used to detect static shadowing on a photovoltaic module?

**Comment by mikofski** [comment]
Hi Naveen,

Thanks for starting this discussion. I think this is a very interesting question. I'm curious to hear what others have to say on this topic. I think it will depend on a number of factors:
* central or string inverter
* multi max-power-point-tracking (MPPT) inverter or single MPPT inverters
* how many strings per MPPT
* crystalline silicon (cSi) or thin film
* twin, tiled, or full size cSi modules
* how many bypass diodes
* What is the voltage operating range of the MPPT in the inverter
* how much shade

In my opinion, if you have a single central inverter connected to many strings of standard full-size cSi modules (not split in half), and there is one string with a module (or modules) that have more than about 10% of static shade on it (them), for example the bottom row is shaded, then you should have significant impact, for example ~90% loss of power, on that string. It doesn't matter if it's only one module or two, or the entire string. This is because the string has to operate at the voltage of all the other presumably unshaded strings in the system, as determined by the inverter. Therefore the worst cell in the string will determine the current for the entire string. For a more detailed explanation, see my post on [why electrical mismatch losses are unintuitive](https://breakingbytes.github.io/electric-mismatch-in-silicon-cell-pv-is-not-intuitive.html#electric-mismatch-in-silicon-cell-pv-is-not-intuitive).

For twin modules, you should see half that loss, because the top and bottom are in parallel. For example if the bottom half has >10% shade, then the string will lose about 40-50% loss.

For thin films shade, I think you will probably see a similar high loss for any amount of non-uniform shade on a single string.

For full-size cSi systems using multi-MPPT inverters or string inverters where there's only one string per inverter or MPPT, then the bypass diodes should turn on and you should only see losses proportional to the number of bypass diodes that are triggered. For example, in a string of 20 modules, each with 3 bypass diodes, if one module and part of its neighbor is shaded >10%, then you might see 3 to 5 out of the 60 bypass diodes activated, so you might only see about a 5-8% loss.

Now to answer your question:

>How can monitoring data be used to detect static shadowing on a photovoltaic module?

I would compare simulated output and look for negative deltas in the monitoring data that are approximately the value of one string. For example in a system with 10 strings on each inverter, if you see a 10% power loss from the model, and other times the measurements match the model, then it may be shade.

I think you will have challenges with systems with twin modules or if any bypass diodes are activated because the losses will be so small, and they might be masked by other losses or variability/measurement noise.

**Reply by Naveen-learner** [reply]
Hello Mark,
Thank you for taking the time to leave a comment.
I'm trying to figure out whether there's any static shading, such as bird droppings, and I only have data from measured current and voltage. As you mentioned, comparing simulated output will detect shade, but how can we tell if it's static or not, and if it's static, how can we tell if it's due to chimney, leave falling, or other factors? I'd want to model all of these aspects in Python; could you perhaps give me some ideas or point me in the right direction?
Thank you in advance.

**Reply by cwhanse** [reply]
Naveen, as a rough guideline, shading from objects on the module's surface will affect current and voltage at all times, whereas shading from nearby structures such as a tree or pole will affect the current and voltage only while the sun is behind the structure. If you plot power on a 2D image, with time of day on one axis, and day of year on the other, you can often "see" the nearby structure in the image.

**Reply by mikofski** [reply]
Wow! That is so cool! Would you see it even better if you mapped time of day/year to azimuth & zenith, and plotted it that way?

**Reply by AdamRJensen** [reply]
Zenith vs azimuth is indeed better. For direct irradiance measurements I typically use groupby on azimuth and zenith in 1 degree bins and take the maximum value. Taking the maximum value serves to filter out low irradiance periods. Plotting a 2D plot then clearly illustrates shading by trees and other objects

**Reply by cwhanse** [reply]
Plotting in 2D (time vs. day of year) distorts the object's shape, plotting by zenith and azimuth tends to preserve it. I'm usually too lazy to bin the data before plotting :)

**Reply by Naveen-learner** [reply]
> Zenith vs azimuth is indeed better. For direct irradiance measurements I typically use groupby on azimuth and zenith in 1 degree bins and take the maximum value. Taking the maximum value serves to filter out low irradiance periods. Plotting a 2D plot then clearly illustrates shading by trees and other objects

Could you please elaborate further and, if possible, provide the Python code?

**Reply by Naveen-learner** [reply]
> Naveen, as a rough guideline, shading from objects on the module's surface will affect current and voltage at all times, whereas shading from nearby structures such as a tree or pole will affect the current and voltage only while the sun is behind the structure. If you plot power on a 2D image, with time of day on one axis, and day of year on the other, you can often "see" the nearby structure in the image.

I can see just the shading that occurred by plotting the 2D graph, but how can I identify the objects?

**Reply by Naveen-learner** [reply]
> Wow! That is so cool! Would you see it even better if you mapped time of day/year to azimuth & zenith, and plotted it that way?

I tried plotting azimuth and zenith angles with PVLIB, but I'm not sure how to plot shading in those graphs.
